### PR TITLE
Dbz 5804 correct mongodb incremental snapshot instructions

### PR DESCRIPTION
**Pull request by roldanbob**
[DBZ-5804](https://issues.redhat.com/browse/DBZ-5804)

The instructions for triggering and stopping an incremental snapshot are contained in a file that is shared across all of the connectors. The instructions are SQL-based and do not apply to MongoDB. This change refactors the existing content:
It breaks the procedures for triggering a snapshot and stopping a snapshot into separate files.
It adds instructions about how to insert a document into a MongoDB signaling collection.
It tags the SQL and NoSQL instructions in the shared file and then uses  the new tag attributes in the include directives in the connector files.

Tested in a local Antora build. 

Additional changes are also likely to be needed in `signalling.adoc` to make the content more generally applicable.  

Please backport this change to 2.0.
I think that we should probably also backport  to 1.9, yes?

### CODE REVIEW

File: documentation/modules/ROOT/partials/modules/all-connectors/proc-stopping-an-incremental-snapshot.adoc [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    '{"data-collections": ["schema1.table1", "schema2.table2"], // <4>
+    "type":"incremental"}'); // <5>
+----
+end::sql-based-snapshot[]
+tag::nosql-based-snapshot[]
+. Insert a stop snapshot signal document into the signaling {data-collection}:
++
+[source,bash,indent=0,subs="+attributes,+quotes"]
+----
+_<signalDataCollection>_.insert({"id" : _<idNumber>_,"type" : "stop-snapshot", "data" : {"data-collections" ["<collectionName", "collectionName"],"type": "incremental"}});
```
Comment (Naros) [inline documentation/modules/ROOT/partials/modules/all-connectors/proc-stopping-an-incremental-snapshot.adoc:44]:
```suggestion
_<signalDataCollection>_.insert({"id" : _<idNumber>_,"type" : "stop-snapshot", "data" : {"data-collections" ["_<collectionName>_", "_<collectionName>_"],"type": "incremental"}});
```

File: documentation/modules/ROOT/partials/modules/all-connectors/proc-triggering-an-incremental-snapshot.adoc [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+end::nosql-based-snapshot[]
+
+The `data-collections` array for an incremental snapshot signal has no default value.
+If the `data-collections` array is empty, {prodname} detects that no action is required and does not perform a snapshot.
+
+[NOTE]
+====
+If the name of a {data-collection} that you want to include in a snapshot contains a dot (`.`) in the name of the database, schema, or table, to add the {data-collection} to the `data-collections` array, you must escape each part of the name in double quotes. +
+ +
+For example, to include a table that exists in the `*public*` schema and that has the name `*My.Table*`, use the following format: `*"public"."My.Table"*`.
```
Comment (Naros) [inline documentation/modules/ROOT/partials/modules/all-connectors/proc-triggering-an-incremental-snapshot.adoc:29]: We may want to conditionalize even this with a NoSQL vs SQL bit just due to the wording used.

File: documentation/modules/ROOT/partials/modules/all-connectors/proc-triggering-an-incremental-snapshot.adoc [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+----
+end::sql-based-snapshot[]
+
+tag::nosql-based-snapshot[]
+
+. Insert a snapshot signal document into the signaling {data-collection}:
++
+[source,bash,indent=0,subs="+attributes,+quotes"]
+----
+_<signalDataCollection>_.insert({"_id" : _<idNumber>_,"type" : _<snapshotType>, "data" : {"data-collections" ["<collectionName", "collectionName"],"type": _<snapshotType>_}});
```
Comment (Naros) [inline documentation/modules/ROOT/partials/modules/all-connectors/proc-triggering-an-incremental-snapshot.adoc:68]:
```suggestion
_<signalDataCollection>_.insert({"_id" : _<idNumber>_,"type" : _<snapshotType>_, "data" : {"data-collections" ["_<collectionName>_", "_<collectionName>_"],"type": _<snapshotType>_}});
```

**Review by Naros — CHANGES_REQUESTED** [review]
Hi @roldanbob, I found a few small things I believe we need to fix before we can merge this.

File: documentation/modules/ROOT/partials/modules/all-connectors/proc-triggering-an-incremental-snapshot.adoc (hunk 2 above)
Comment (roldanbob) [reply documentation/modules/ROOT/partials/modules/all-connectors/proc-triggering-an-incremental-snapshot.adoc:29]: Took me a few reads to understand which "this" you were referring to. The note, right? I'll make the necessary changes to that. thanks.

Comment (roldanbob) [inline documentation/modules/ROOT/partials/modules/all-connectors/proc-triggering-an-incremental-snapshot.adoc:29]:
```suggestion
tag::sql-based-snapshot[]
For example, to include a table that exists in the `*public*` schema and that has the name `*My.Table*`, use the following format: `*"public"."My.Table"*`.
end::sql-based-snapshot[]
tag::nosql-based-snapshot[]
For example, to include a data collection that exists in the `*public*` database, and that has the name `*MyCollection*`, use the following format: `*"public"."MyCollection"*`.
end::nosql-based-snapshot[]
```

**Review by roldanbob — COMMENTED** [review]
Updated the content per your suggestions, @Naros

File: documentation/modules/ROOT/partials/modules/all-connectors/proc-stopping-an-incremental-snapshot.adoc [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+    '{"data-collections": ["schema1.table1", "schema2.table2"], // <4>
+    "type":"incremental"}'); // <5>
+----
+end::sql-based-snapshot[]
+tag::nosql-based-snapshot[]
+. Insert a stop snapshot signal document into the signaling {data-collection}:
++
+[source,bash,indent=0,subs="+attributes,+quotes"]
+----
+_<signalDataCollection>_.insert({"id" : _<idNumber>_,"type" : "stop-snapshot", "data" : {"data-collections" ["_<collectionName>_", "_<collectionName>_"],"type": "incremental"}});
```
Comment (jpechane) [inline documentation/modules/ROOT/partials/modules/all-connectors/proc-stopping-an-incremental-snapshot.adoc:44]:
```suggestion
_<signalDataCollection>_.insert({"_id" : _<idNumber>_,"type" : "stop-snapshot", "data" : {"data-collections" ["_<collectionName>_", "_<collectionName>_"],"type": "incremental"}});
```

**Comment by roldanbob** [comment]
> @roldanbob There is a small typo left and rebase is needed. When corrected the PR can be merged. Thanks a lot!

Fixed typo and rebased.

**Comment by jpechane** [comment]
@roldanbob Applied, thanks!

**Comment by jpechane** [comment]
@roldanbob Applied, unfortunately this does not apply cleanly on 2.0 or 1.9 :-(

Could you please prepare a separate PRs with different base branches? Thanks a lot

**Pull request merged**